src/algebra/order/to_interval_mod.lean

```diff
@@ -733,3 +733,78 @@ lemma to_Ico_mod_zero_one (x : α) : to_Ico_mod (0 : α) zero_lt_one x = int.fra
 by simp [to_Ico_mod_eq_add_fract_mul]
 
 end linear_ordered_field
+
+section union
```

Comment:
Can you prove all corresponding lemmas about `Ico`?

Reply:
Done at #18427 (along with `Icc` and `Ioo` cases where they hold).